proof/refine/RISCV64/Ipc_R.thy

```diff
@@ -5461,26 +5476,25 @@ lemma receiveIPC_corres:
                                                   and valid_sched_pointers
                                                   and valid_objs'
                                                   and valid_bound_obj' valid_replies'_sc_asrt replyOpt
+                                                  and valid_bound_reply' replyOpt
                                                   and pspace_aligned' and pspace_distinct'
                                                   and pspace_bounded'
                                                   and (\<lambda>s. weak_sch_act_wf (ksSchedulerAction s) s)"
                                            in corres_guard_imp [OF corres_if])
                                   apply (simp add: fault_rel_optionation_def)
                                  apply (rule corres_if2)
                                    apply simp
-                                  apply (rule corres_split_eqr[OF threadGet_corres replyPush_corres])
+                                  apply (rule corres_split_eqr[OF threadGet_corres])
                                      apply (clarsimp simp: tcb_relation_def)
-                                    apply (clarsimp simp: fault_rel_optionation_def split: option.splits)
-                                   prefer 3 \<comment> \<open> defer wp until corres complete \<close>
-                                   apply (rule setThreadState_corres, simp)
-                                  prefer 3 \<comment> \<open> defer wp until corres complete \<close>
-                                  apply (rule corres_split[OF setThreadState_corres], simp)
-                                    apply (rule possibleSwitchTo_corres, simp)
-                                   apply (wpsimp wp: set_thread_state_valid_sched_action)
-                                  apply wpsimp
-                                 apply wpsimp
-                                apply wpsimp
-                               apply clarsimp
+                                    apply (rule corres_split[OF replyCanGrant_update_corres replyPush_corres])
+                                       apply simp
+                                      apply (clarsimp simp: fault_rel_optionation_def split: option.splits)
+                                     prefer 5 \<comment> \<open> defer wp until corres complete \<close>
```

Comment:
Could this `prefer` be done instead with a `match` of some kind? `match premisses` or whatever it is?